### ISSUE TITLE
Feature: mp4 support

### ISSUE THREAD

**Issue by hobgoblina** [FredHappyface/Android.EweSticker#80]
## Before You Begin

- [x] I have checked for similar feature requests in the project's issue tracker
  to avoid duplicates.
- [x] I have searched existing issues to see if this feature has been discussed
  before.

## Feature Details

mp4 support would be dope. this was requested in #34 but they didn't try to justify it.

### Is your feature request related to a problem? 
when i try to use mp4 files with ewe, it replaces the video with a static image of the first frame.

mp4 files are much more likely to be high quality compared with gifs.

these days, i'm actually coming across mp4s more often than gifs because of the higher quality. e.g., a significant amount of the files supplied by giphy are mp4s these days.

additionally, it's actually easier to use mp4s with certain services. e.g., on mastodon i often have issues posting gifs and have to convert them to mp4s. this also means that the "gifs" that other people post, which I may want to save and add to my ewe collection, are most likely mp4s.

### Describe the solution you'd like

support mp4 video files. maybe you could impose a file size limit, if that would help make the implementation easier? all the mp4s i'm trying to add are less than half a MB, even when they're ~15 seconds long and fairly high res (compared to gifs at least... no visible artifacts on phones)

### Describe alternatives you've considered

converting to gifs, which is a real pain and results in significantly lower quality unless i go through a significant amount of effort to do it via gimp/etc, which nobody has time for when you just wanna save a reaction gif

**Comment by FredHappyface** [comment]
Thanks so much for the issue. Taken a look at the coil docs (which is the library I use to load images) and it seems to have limited support for MP4 files. Unfortunately it'll be a while before I add this :)

**Comment by FredHappyface** [comment]
MP4 files are already supported if the app supports them. The only app I've tested that does so far is moshidon, so the first frame is taken and passed over as a png. I have made some code improvements behind the scenes to this so ewesticker should be a bit better at sending *something* if the app doesn't support the format

**Comment by hobgoblina** [comment]
i've tried it in multiple apps that support mp4s both via attachments and via clipboard, and it only ever supplies the first frame as a png.

**Comment by hobgoblina** [comment]
also in the ewesticker keyboard there's no animation, unlike gifs. only shows the first frame. so it seems like ewesticker might be coverting the mp4s to png during import

**Comment by FredHappyface** [comment]
Yeah so what EweSticker does is push the sticker over a "InputConnectionCompat" to an application like discord or whatever, that app is responsible for providing EweSticker with a list of mimetypes (file types) that it can support 

Discord says it can support any image (technically not true as svgs do not work with it)
![Image](https://github.com/user-attachments/assets/3e6472a4-e32e-4edd-814e-71ccf29f5317)

Here you can see WhatsApp supports very few images 
![Image](https://github.com/user-attachments/assets/3984c704-c1cf-4b0d-b1e1-4137e45068f9)

Note that all of these apps can likely support way more as attachments, but that's not how EweSticker works. This could be suported via the share sheet which is currently tried last (the assumption is made that a 1-or-2 tap static image is preferable to using the android share menu, which is a bit crap)

You are absolutely correct that a png is sent. The createCompatSticker function is called in the event that the app doesn't support the mime type that you are sending. This creates and keeps a cache of PNGs if required (this means that the first time you send it, it'll take a second or so and in future it should be near-instant) 

I'd be happy to add a setting to always use the share menu if a file type isn't supported if that'd be useful to you? Sorry that I can't help anymore with this - spent hours and this is honestly the best it gets as its completely out of my control what other apps support

**Comment by FredHappyface** [comment]
That said by all means share what apps you are using and I'll take a look, in future you'll also be able to share log files (though you might have to clear out any personal info such as names of stickers yourself in that case)

**Comment by hobgoblina** [comment]
the main apps i'm hoping to use it with are Signal, Cheogram/Monocles, and Fedilab.

yeah, share menu would be better than a png of the first frame. tho for must use cases it's probably less effort to go thru the filesystem to use it as an attachment. oh well. thanks for looking into it!

**Comment by FredHappyface** [comment]
Thanks for this, 

Added a toggle to use the share menu and siscoved a way to make sharing less painful too so when you send an mp4, signal will prompt you to select the chat to send the mp4 in. I plan on releasing in a few hours
